src/solvers/refinement/string_builtin_function.cpp

```diff
@@ -187,6 +187,46 @@ exprt string_set_char_builtin_functiont::length_constraint() const
     equal_exprt(return_code, return_value));
 }
 
+static bool eval_is_upper_case(const mp_integer &c)
+{
+  return ('A' <= c && c <= 'Z') || (0xc0 <= c && c <= 0xd6) ||
+         (0xd8 <= c && c <= 0xde);
```

Comment:
This says to me we're not doing ASCII.  Might you add a comment to say where these numbers come from?  Also, is there a reason to not use isupper et al.?

Reply:
Yes I documented that so that it should be clearer now. We do a bit more that ASCII but not all unicode (that dould be quite complicated indeed). My idea is that we will provide this builtin function accessible as `CProverString.toUpperCase`, which only coincide with String.toUpperCase for a fragment of unicode (basic latin and latin-1 supplement for now), you can then provide a model of String.toUpperCase which either over-approximate (useful for verification) or under-approximate (useful for test generation) the behaviour of the JDK function.

Reply:
This sounds scary. Please add tests and documentation (really, both!) that very precisely document the boundaries of what is/is not covered. I accept that there may not be enough time to do a perfect job (cover all of unicode), but then there needs to be clear documentation of what has (not) been done to avoid surprises.

Reply:
Yes; please, what @tautschnig said.